sdcm/nemesis.py

```diff
@@ -1830,14 +1830,24 @@ def repair_nodetool_rebuild(self):
             self.target_node.run_nodetool('rebuild', retry=0)
 
     def disrupt_nodetool_cleanup(self):
-        # This fix important when just user profile is run in the test and "keyspace1" doesn't exist.
-        test_keyspaces = self.cluster.get_test_keyspaces()
-        for node in self.cluster.nodes:
+        # Inner disrupt function for ParallelObject
+        def _dis_nodetool_cleanup(node):
+            # This fix important when just user profile is run in the test and "keyspace1" doesn't exist.
+            test_keyspaces = self.cluster.get_test_keyspaces()
             InfoEvent('NodetoolCleanupMonkey %s' % node).publish()
             with adaptive_timeout(Operations.CLEANUP, node, timeout=HOUR_IN_SEC * 48):
                 for keyspace in test_keyspaces:
                     node.run_nodetool(sub_cmd="cleanup", args=keyspace)
 
+        results = ParallelObject(self.cluster.nodes, num_workers=min(32, len(self.cluster.nodes))).\
+            run(_dis_nodetool_cleanup, ignore_exceptions=True, unpack_objects=True)
+        runs_that_finished_with_exception = [res for res in results if res.exc]
+        if runs_that_finished_with_exception:
```

Comment:
Use generator to avoid processing of redundant elements:
```
if any(thread_result for thread_result in thread_results if thread_result.exc):
    ...
else:
    ...
```

Reply:
Changed.

The same code is present in common.py lines 436-439.
`runs_that_finished_with_exception = [res for res in results if res.exc]
        if runs_that_finished_with_exception:
            raise ParallelObjectException(results=results)
        return results`
Should we change it there as well?

Reply:
ok